packages/ui/src/components/input-field/input-field.tsx

```diff
@@ -0,0 +1,74 @@
+import clsx from 'clsx';
+import React from 'react';
+
+import { Text } from '../text';
+import { SelectInputProps } from './select-input';
+import { TextInputProps } from './text-input';
+
+export type InputFieldProps = {
+  children: React.ReactElement<TextInputProps | SelectInputProps>;
+  label?: string;
+  placeholder?: string;
+  errorMessage?: string;
+  hintText?: string;
+  disabled?: boolean;
+  className?: string;
+  /**
+   * @default 'vertical'
+   */
+  layout?: 'vertical' | 'horizontal';
+};
+
+export function InputField({
+  label,
+  errorMessage,
+  hintText,
+  children,
+  layout = 'vertical',
+  className,
+  ...childrenProps
+}: InputFieldProps): JSX.Element {
+  return layout === 'vertical' ? (
+    <label
+      className={clsx(
+        'flex h-full w-full flex-col items-start gap-1.5',
+        className,
+      )}
+    >
+      <p className="text-sm font-medium">{label}</p>
+      {React.cloneElement(children, {
+        error: !!errorMessage,
+        ...childrenProps,
+      })}
+      {errorMessage && (
+        <Text role="alert" variant="error" size="xs">
+          {errorMessage}
+        </Text>
+      )}
+      {!errorMessage && hintText && (
```

Comment:
We only show hintText if there is no errorMessage?

Reply:
OK, looks like this is the right behvior from design

Reply:
Maybe add <AnimationPresent> here for animations

Reply:
> Maybe add AnimatePresence here for animations

Good point. I'll try adding it.

Reply:
@devrsi0n Added. Please help review again.